src/metabase/notification/payload/impl/system_event.clj

```diff
@@ -126,6 +128,25 @@
                             :status "discontinued"}}
      :map]]])
 
+(defn- normalize-record
+  [record field-names]
+  ;; handle different casing from different driver
+  (let [field-names      (set field-names)
+        col->field-name  (merge (u/for-map [f field-names]
+                                  [(u/lower-case-en f) f])
+                                (u/for-map [f field-names]
+                                  [(u/upper-case-en f) f]))]
+    (update-keys (update-keys record u/qualified-name) col->field-name)))
+
+(defn- ordered-record-map
+  [record ordered-fields]
```

Comment:
nit: I think swapping the argument order is more natural.

1. Practicality - you can partially apply it to map over multiple records.
2. Aesthetics - when calling it with a dynamically constructed record, the multi-line argument comes second.